src/components/FileUpload.tsx

```diff
@@ -0,0 +1,107 @@
+import { css, cx } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { IconButton } from '@grafana/ui';
+import { ChangeEvent, DragEvent, ReactNode, useCallback, useRef, useState } from 'react';
+
+import { useStyles } from '../theme';
+import { H4 } from './Text/TextElements';
+
+interface FileUploadProps {
+  children: ReactNode;
+  title: string;
+  onInputRead: (input: string) => void;
+  accept: string;
+}
+
+export const FileUpload = ({ children, title, onInputRead, accept }: FileUploadProps) => {
+  const [isDragOver, setIsDragOver] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const styles = useStyles(getStyles);
+
+  const handleUpload = useCallback(
+    (files: FileList) => {
+      const file = files[0];
+
+      const acceptedExtensions = accept.split(','); // Split the accept string into an array
+      const extension = file.name.split('.').pop() || ''; // Get file extension
+
+      // Check if the file has an accepted extension
+      if (!acceptedExtensions.includes(extension)) {
+        alert('Unsupported file type');
+        return;
+      }
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        const text = e.target?.result;
+        if (typeof text === 'string') {
+          onInputRead(text);
+        } else {
+          throw Error('Unsupported file type');
+        }
+      };
+      reader.readAsText(file);
+    },
+    [onInputRead, accept]
+  );
+
+  const onFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const files = e.target.files;
+    if (files) {
```

Comment:
Nit; As we directly access the first element (`const file = files[0];`), should we also check here that files is not empty? 🤔